update digitalOcean cloud-init datasource options

jurraca opened the pull request:
Unless I'm missing something, the `datasource` and `datasource_list` options don't exist for `cloud-init`, and cause the evaluation of the flake to fail. 
Tested with local flake eval and deploy to digital ocean.
See https://search.nixos.org/options?channel=unstable&query=cloud-init

eyJhb commented:
They are present in the documentation here https://cloudinit.readthedocs.io/en/latest/reference/base_config_reference.html#datasource-list , most likely it should have been under settings. But I didn't get any error when I ran it last time AFAIK. But as long as this is tested, and confirmed to be working without manually setting an IP then I'm all for it. :)

jurraca commented:
to test, you can run a `flake check` with only the `nixosConfigurations.digitalocean` attr with the `cloud-init-datasource` options there, and it should fail. 
I've had trouble with the cloud-init service on DO and still haven't figured out a working solution, will try with `settings`, thanks.

jurraca commented:
Updated to add the options under `settings`, and added the correct datasource, which seems to be `ConfigDrive` according to the deprecations [here](https://cloudinit.readthedocs.io/en/latest/reference/datasources/digitalocean.html). Current stable and unstable `cloud-init` versions are 24.2, so should be ok.
